### PR TITLE
Update stability tests

### PR DESCRIPTION
**Pull request by yunchu**
### Summary

Update weekly workflow to run fuzzing with different extra installation option instead of pytest-loop.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: .github/workflows/weekly_check.yml [hunk 1]
```diff
@@ -19,17 +19,17 @@ jobs:
...
-          - os: "ubuntu-20.04"
-            tox-env-os: "lin"
-          - os: "windows-2022"
-            tox-env-os: "win"
-    name: stability test (${{ matrix.os }}, Python ${{ matrix.python-version }})
-    runs-on: ${{ matrix.os }}
+          - python-version: "3.11"
+            tox-env-py: "311"
+    name: stability test (Python ${{ matrix.python-version }}, Extra ${{ matrix.extra }})
+    runs-on: ubuntu-latest
```
Comment (wonjuleee) [inline .github/workflows/weekly_check.yml:32]: can I ask you why testing on different `os` is dropped?

File: tox.ini [hunk 2]
```diff
@@ -76,29 +76,30 @@ commands =
...
 allowlist_externals =
     bash
 commands_pre =
     bash -c 'cargo -V; echo "cargo (rust) version checking exit code = $?"'
-    python -m pip uninstall pytest-stress -y
 commands =
-    python -m pytest -v --csv={toxworkdir}/results-{envname}.csv -x {posargs:--loop 5}
-
+    ; coverage erase
+    python tests/fuzzing/cli_fuzzing.py {posargs:-dict=tests/fuzzing/assets/cli_operations.dict -artifact_prefix={toxworkdir}/ -print_final_stats=1 -atheris_runs=500000}
```
Comment (wonjuleee) [inline tox.ini:89]: what is the difference between fuzzing and stability tests?

File: .github/workflows/weekly_check.yml (hunk 1 above)
Comment (yunchu) [reply .github/workflows/weekly_check.yml:32]: we're using fuzzing tool `atheris` but they does not support windows.

File: tox.ini (hunk 2 above)
Comment (yunchu) [reply tox.ini:89]: Stability tests repeatedly runs all tests 5 times sequentially and the results were used for the stress test report which is a requirement to the release but it was replaced with the fuzzing after we implemented it.
Fuzzing or fuzz testing is the technic to verify stability or robustness of interfaces such as CLI or API that exposed from the project and it works by passing randomly generated data to the CLI or API as the arguments.
The expectation of this changes would be that datumaro CLI provides proper help messages or usage without having any uncoughted exceptions on the different environments that setup by using different extra options while installing datumaro.

File: .github/workflows/weekly_check.yml (hunk 1 above)
Comment (wonjuleee) [reply .github/workflows/weekly_check.yml:32]: Is this okay to skip tests for windows?

File: tox.ini (hunk 2 above)
Comment (wonjuleee) [reply tox.ini:89]: Is there any background of having `atheris`?

Comment (wonjuleee) [reply tox.ini:89]: Previously, I meant that fuzzing requires 500K runs, while stability runs the same test code only for 5K times.
Is stability for testing extra dependencies? Could you let me know the test schedule for them (daily, weekly, manual, etc.)?

File: .github/workflows/weekly_check.yml (hunk 1 above)
Comment (yunchu) [reply .github/workflows/weekly_check.yml:32]: current fuzzing implementation is very primitive and works only with datumaro CLI. but I can say that this changes would cover small portion of stability tests that required to this product. it's better to work as well for the windows but it's not supported by a tool of fuzzing we're using.

File: tox.ini (hunk 2 above)
Comment (yunchu) [reply tox.ini:89]: fuzzing will be performed once per a release (not scheduled, on demand base) but I want to verify basic stability of the datumaro CLI on all available installation environments of datumaro those can be created through the different extra options. To do that, i don't think that needs to attempt too much trials and it's good to run with weekly schedule.

**Review by wonjuleee — APPROVED** [review]

**Pull request merged**